### PR TITLE
[Tidy] Remove redundant template chart configurations

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Remove redundant template configurations (e.g. go.Table, bar specific configurations)

Note: Based on this issue https://github.com/mckinsey/vizro/issues/673 I've initially raised, it's important to understand  that `template.data.XXX` impacts both go.Figures and px charts. However, we configured it incorrectly, which is why our specified bar width didn't take effect. However, while creating this PR, it became clear that we should not set a default bar width: https://github.com/mckinsey/vizro/pull/679#issuecomment-2329614756

Another point to note is that px charts can override certain settings from the template, such as the` legend_title_text`. From my observations, they also override the titles and values of the x/y axes, the legend title and the overall color sequence. This means that any specifications within the template unfortunately don't apply. Consequently, I've removed the configuration for the legend title since it had no effect (by default, it's turned off for go.Figures).

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/src/vizro/_themes/_templates/template_dark.py [hunk 1]
```diff
@@ -71,31 +71,7 @@ def create_template_dark() -> Template:
...
 
     # CHART TYPES
-    template_dark.data.bar = [
-        go.Bar(
-            marker={"line": {"color": template_dark["layout"]["paper_bgcolor"], "width": 1}},
-            textfont={"color": COLORS["WHITE_55"]},
-        )
-    ]
-
-    template_dark.data.table = [
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/_themes/_templates/template_dark.py:81]: Removed the go.Table as this was always just intended as a temporary solution until we had the dash_data_table and/or dash_ag_grid in place

File: vizro-core/src/vizro/_themes/_templates/common_values.py [hunk 2]
```diff
@@ -40,7 +40,6 @@ def create_template_common():
         legend_y=-0.20,
         legend_title_font_family=primary_font,
         legend_title_font_size=14,
-        legend_title_text="",
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/_themes/_templates/common_values.py:43]: Doesn't have any effect, therefore removing here

**Review by petar-qb — APPROVED** [review]
I like it! 🧹

**Comment by petar-qb** [comment]
For reviewers: 

This is the example from the `main` branch:
image

And this is from the feature branch:
image

**Comment by huong-li-nguyen** [comment]
Seems like that change on the bar width doesn't work well on the `px.bar` if the data is not aggregated. If I can't get both working, I might have to undo the bar width change 😢 , but the template removal will stay

Screenshot 2024-09-04 at 18 16 41

**Comment by huong-li-nguyen** [comment]
Okay, I went into the go.Bar source code and it seems like we shouldn't set the bar.width inside the template. What previously worked fine for a categorical x-axis, doesn't work anymore for a numerical x-axis, and it's because the width is defined in "position axis units" so needs to vary depending on the x-axis scale. Fixing it inside would therefore only look good on a categorical axis, but not on a numeric one. Will therefore remove the setting of the bar width, but the rest stays 👍 

Screenshot 2024-09-04 at 18 21 56

**Review by nadijagraca — APPROVED** [review]
LGTM. 👍

**Pull request merged**